### DISCUSSION TITLE
Errore Nest Thermostat

### DISCUSSION THREAD

**Discussion by fabioantoci** [indomus/forum#2052] [HUB: Home Assistant]
Ciao a tutti

ho aggiornato HA Core all'ultima release 2022.11.1 aggiornando anche Python alla 3.10.
Dopo l'aggiornamento non mi funziona più l'integrazione con il termostato Nest.

Il log riporta il seguente errore:
`Setup failed for nest: Unable to import component: /lib/arm-linux-gnueabihf/libm.so.6: version 'GLIBC_2.29' not found (required by /srv/homeassistant/lib/python3.10/site-packages/grpc/_cython/cygrpc.cpython-310-arm-linux-gnueabihf.so)`

Avete idee sul come risolvere?
Grazie

**Comment by Scialla** [comment]
Che OS hai? Che versione?
Python 3.10 non è supportato su vecchi OS

<html>
<body>


Raspian/debian nickname | number | python support
-- | -- | --
Buster | 10 | Python 3.7
Bullseye | 11 | Python 3.9


</body>
</html>

vedi ad esempio qui, ma puoi cercarne altri:
https://github.com/numpy/numpy/issues/20371

Consiglio: io ho smesso di litigare su ste cose passando a HA core con docker.. e vivo felice (anche se Venv era la Via) :D

**Comment by fabioantoci** [comment]
Ciao Scialla scusami per il ritardo nella risposta ma sono stato bloccato da una serie di circostanze.
Grazie per la risposta!
Dopo aver aggiornato il sistema operativo ho dato per scontato fosse supportata la versione 3.10 di Python.
Attualmente ho installato Bullseye con Kernel 5.10.103.

Migrare tutto su Docker è una cosa che avevo intenzione di fare a breve, procedo in questo modo e vedo se riscontro problemi.
Grazie ancora

Venv è la via ma Docker è la panacea di tutti (molti) mali :D

**Comment by fabioantoci** [comment]
Ho riscontrato un problema che non mi fa aggiungere il termostato Nest ad HA.
Ho installato HA su Docker. Accedo dall'esterno della mia rete tramite reverse proxy (Caddy - duckdns).

Il processo di configurazione del termostato procede correttamente compreso l'accesso all'account gmail.

Effettuo il link del mio account ad HomeAssistant
![Schermata](https://user-images.githubusercontent.com/75075472/204135169-44c1fd1b-fcec-43d1-a989-204674e95dd9.png)

ma ottengo un errore sul pannello di configurazione delle integrazioni
![Schermata2](https://user-images.githubusercontent.com/75075472/204135219-cd2bd06f-20d9-45de-b5cd-3066493bffd5.png)

il log riporta:
```ERROR (MainThread) [homeassistant.components.nest.config_flow] Error creating subscription: Failed to create subscriber 'projects/mio_id_progetto/subscriptions/home-assistant-CjySl1tR3J': Deadline of 60.0s exceeded while calling target function, last exception: 503 DNS resolution failed for pubsub.googleapis.com:443: C-ares status is not ARES_SUCCESS qtype=AAAA name=pubsub.googleapis.com is_balancer=0: Could not contact DNS servers```

Come si può risolvere? Avete idee?

**Comment by fabioantoci** [comment]
A qualcuno è successa una cosa simile?
Qualcuno ha un suggerimento su come procedere?
Sto sbagliando a configurare qualcosa?

Grazie
